pyrokinetics/gk_code/GKOutputReaderGENE.py

```diff
@@ -128,6 +137,27 @@ def _init_dataset(
         """
         nml = gk_input.data
 
+        pyro_coord_units = coord_units(local_norm.pyrokinetics)
+        if nml["geometry"].get("major_R", 1.0) == 1.0:
+            local_norm.default_convention = "gene"
+        else:
+            try:
```

Comment:
This whole bit is probably complex and distracting enough that I would pull it out into a helper function, something like `set_default_convention()`